### PR TITLE
Add Helper Functions in StarTreeBuilderUtils and StarTreeV2BuilderConfig

### PR DESCRIPTION
**Pull request by aishikbh**
Added the following helper functions:

- generateBuilderConfigs : Generates builder configs from JsonNode segment metadata
- generateDefaultConfig: Generates default config from JsonNode columnsMetadata
- areStarTreeBuilderConfigsEqual: compares if 2 builderConfig lists are equal

### CODE REVIEW

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java [hunk 1]
```diff
@@ -83,6 +85,27 @@ public static List<StarTreeV2BuilderConfig> generateBuilderConfigs(@Nullable Lis
     return builderConfigs;
   }
 
+  public static List<StarTreeV2BuilderConfig> generateBuilderConfigs(@Nullable List<StarTreeIndexConfig> indexConfigs,
+      boolean enableDefaultStarTree, Schema schema, JsonNode segmentMetadata) {
+    List<StarTreeV2BuilderConfig> builderConfigs = new ArrayList<>();
+    if (indexConfigs != null) {
+      for (StarTreeIndexConfig indexConfig : indexConfigs) {
+        StarTreeV2BuilderConfig builderConfig = StarTreeV2BuilderConfig.fromIndexConfig(indexConfig);
+        if (!builderConfigs.contains(builderConfig)) {
```
Comment (snleee) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java:94]: `StarTreeV2BuilderConfig` <- can we double check the `equals()` function for this class?

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java [hunk 2]
```diff
@@ -83,6 +85,27 @@ public static List<StarTreeV2BuilderConfig> generateBuilderConfigs(@Nullable Lis
...
+          builderConfigs.add(builderConfig);
+        }
+      }
+    }
+    if (enableDefaultStarTree) {
```
Comment (snleee) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java:99]: What's the current behavior if

1. custom startree config is added
2. enableDefaultStarTree is set to true

Are we adding default startree or default startree + custom startreee?

**Review by snleee — APPROVED** [review]
LGTM otherwise.

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java (hunk 2 above)
Comment (aishikbh) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java:99]: yes that is the current behaviour right now.

Existing `generateBuilderConfigs(@Nullable List<StarTreeIndexConfig> indexConfigs,
      boolean enableDefaultStarTree, SegmentMetadata segmentMetadata)` hints at that as well.

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java (hunk 1 above)
Comment (aishikbh) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java:94]: `equals()` is properly overridden properly : [here](https://github.com/apache/pinot/blob/5c81c0a88a0a95e5bf150363bb3ed5b5203de795/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/StarTreeV2BuilderConfig.java#L245-L256)

**Pull request merged**